packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebSettingsCompatProxyApi.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.webviewflutter;
+
+import android.webkit.WebSettings;
+import androidx.annotation.NonNull;
+import androidx.webkit.WebSettingsCompat;
+import androidx.webkit.WebViewFeature;
+
+/**
+ * Proxy API implementation for {@link WebSettingsCompat}.
+ *
+ * <p>This class may handle instantiating and adding native object instances that are attached to a
+ * Dart instance or handle method calls on the associated native class or an instance of the class.
+ */
+public class WebSettingsCompatProxyApi extends PigeonApiWebSettingsCompat {
+  public WebSettingsCompatProxyApi(@NonNull ProxyApiRegistrar pigeonRegistrar) {
+    super(pigeonRegistrar);
+  }
+
+  @Override
+  public void setPaymentRequestEnabled(@NonNull WebSettings webSettings, boolean enabled) {
+    if (WebViewFeature.isFeatureSupported(WebViewFeature.PAYMENT_REQUEST)) {
```

Comment:
Why is this check being done internally on the native side, rather than being a straight pass-through? I would expect this to work the same way the SDK works, which is that clients are responsible for this.

Reply:
@stuartmorgan-g This was to address Android lint's RequiresFeature, but as you pointed out, since it's natural to use it after checking `WebViewFeature.isFeatureSupported`, it was indeed excessive checking. Therefore, I will address it with `SuppressLint`.

Reply:
I've added a section about the payment request API usage to the README.